src/WireMock.Net/Server/WireMockServer.Admin.cs

```diff
@@ -555,11 +555,63 @@ private ResponseMessage MappingsPost(RequestMessage requestMessage)
 
         private ResponseMessage MappingsDelete(RequestMessage requestMessage)
         {
-            ResetMappings();
+            // if we allow body for HTTP Delete & a body is defined
+            if (_settings.AllowBodyForAllHttpMethods.HasValue && _settings.AllowBodyForAllHttpMethods.Value && !String.IsNullOrEmpty(requestMessage.Body))
+            {
+                List<Guid> deletedGuids = new List<Guid>();
+                bool success = MappingsDeleteMappingFromBody(requestMessage, ref deletedGuids);
+                if (success)
+                {
+                    return ResponseMessageBuilder.Create($"Mappings deleted. Affected GUIDs: [{String.Join(", ", deletedGuids?.ToArray())}]");
+                }
+                else
+                {
+                    // return bad request
+                    return ResponseMessageBuilder.Create($"Poorly formed mapping JSON.", 400);
+                }
+            }
+            else
+            {
+                ResetMappings();
 
-            ResetScenarios();
+                ResetScenarios();
+
+                return ResponseMessageBuilder.Create("Mappings deleted");
+            }
+        }
+
+        private bool MappingsDeleteMappingFromBody(RequestMessage requestMessage, ref List<Guid> deletedGuids)
```

Comment:
Return `ICollection<Guid>` or  `IEnumerable<Guid>` in case it's ok.
Else return `null` when something throws.